src/test/java/io/cdap/plugin/ariba/source/AribaBatchSourceTest.java

```diff
@@ -81,4 +99,482 @@ public void testConfigurePipelineWithInvalidBasicParam() {
+                                         "https://openapi.au.cloud.ariba.com",
+                                         "prod", "CloudsufiDSAPP-T",
+                                         "SourcingProjectFactSystemView",
+                                         "", "",
+                                         "2022-01-28T10:05:02Z", "2022-01-31T10:05:02Z",
+                                         "serviceAccountJSON");
+    try {
+      aribaBatchSource = new AribaBatchSource(pluginConfig);
+      aribaBatchSource.configurePipeline(pipelineConfigurer);
+    } catch (ValidationException ve) {
+      List<ValidationFailure> failures = ve.getFailures();
+      Assert.assertEquals("Failures size does not match", 3, failures.size());
+      Assert.assertEquals(ResourceConstants.ERR_MISSING_PARAM_PREFIX.getMsgForKey("Ariba Client Id"),
+                          failures.get(0).getMessage());
+
+      Assert.assertEquals(ResourceConstants.ERR_MISSING_PARAM_PREFIX.getMsgForKey("Ariba Client Secret"),
+                          failures.get(1).getMessage());
+    }
+  }
+
+  @Test
+  public void validateMandatoryParameters() {
+    pluginConfig = new AribaPluginConfig("", "",
+                                         "", "",
+                                         "", "clientId",
+                                         "clientSecret", "apiKey",
+                                         "2022-01-28T10:05:02Z", "2022-01-31T10:05:02Z");
+    try {
+      aribaBatchSource = new AribaBatchSource(pluginConfig);
+      aribaBatchSource.configurePipeline(pipelineConfigurer);
+    } catch (ValidationException s) {
+      List<ValidationFailure> failures = s.getFailures();
+      Assert.assertEquals("Failures size does not match", 6, failures.size());
+    }
+  }
+
+  @Test
+  public void testValidateBasicParameters() {
+    Assert.assertEquals("Template does not match", "SourcingProjectFactSystemView",
+                        pluginConfig.getViewTemplateName());
+    Assert.assertEquals("From date does not match", "2022-01-28T10:05:02Z",
+                        pluginConfig.getFromDate());
+    Assert.assertEquals("To date does not match", "2022-01-31T10:05:02Z", pluginConfig.getToDate());
+  }
+
+  @Test
+  public void testValidateCredentialParameters() {
+    pluginConfig = new AribaPluginConfig("referenceName", "baseUrl",
+                                         "prod", "realm",
+                                         "viewTemplateName", "",
+                                         "", "",
+                                         "2022-01-28T10:05:02Z", "2022-01-31T10:05:02Z");
+    try {
+      aribaBatchSource = new AribaBatchSource(pluginConfig);
+      aribaBatchSource.configurePipeline(pipelineConfigurer);
+    } catch (ValidationException s) {
+      List<ValidationFailure> failures = s.getFailures();
+      Assert.assertEquals("Failures size does not match", 3, failures.size());
```

Comment:
why do we expect 3 validation errors? please add comments when test expectation is not obvious. same comment for test cases below.

Reply:
Fixed here (https://github.com/data-integrations/ariba-plugins/pull/2/commits/9e5462f14445ea32445eb3bdd766d3d83f76499b)